File: python/src/core/point3d.py
from point import Point

import math

class Point3D(Point):
	def __init__(self, *args):
		self.dim = 3
		if len(args) == 0:
			self.x = [0, 0, 0]
		if len(args) == 1:
			self.x = args[0]

	def cross_prod(point1, point2 = None, self = None):
		if self != None:
			result = Point(3, [0]*3)
			result.x[0] = self.x[1] * point1.x[2] - self.x[2] * point1.x[1]
			result.x[1] = self.x[2] * point1.x[0] - self.x[0] * point1.x[2]
			result.x[1] = self.x[0] * point1.x[1] - self.x[1] * point1.x[0]
			return result
		else:
			result = Point(3, [0]*3)
			result.x[0] = point1.x[1] * point2.x[2] - point1.x[2] * point2.x[1]
			result.x[1] = point1.x[2] * point2.x[0] - point1.x[0] * point2.x[2]
			result.x[2] = point1.x[0] * point2.x[1] - point1.x[1] * point2.x[0]
			return result

	def mix_prod(point1, point2, point3, self = None):
		if self != None:
			result = self.x[0] * (point1.x[1] * point2.x[2] - point2.x[1] * point1.x[2]) + self.x[1] * (point1.x[2] * point2.x[0] - point2.x[2] * point1.x[0]) + self.x[2] * (point1.x[0] * point2.x[1] - point2.x[0] * point1.x[1])
		else:
			result = point1.x[0] * (point2.x[1] * point3.x[2] - point3.x[1] * point2.x[2]) + point1.x[1] * (point2.x[2] * point3.x[0] - point3.x[2] * point2.x[0]) + point1.x[2] * (point2.x[0] * point3.x[1] - point3.x[0] * point2.x[1])
		return result

p1 = Point3D([1, 1, -2])
p2 = Point3D([2, 2, 3])
p3 = Point3D([2, 3, 1])

print(Point3D.mix_prod(p1, p2, p3))

File: csharp/WorkOOP/Point3D.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WorkOOP
{
    // Реализация класса Point3D – описывает точку в 3-мерном пространстве.
    internal class Point3D : Point
    {
        /*
         * Point3D – конструктор, описывающий инициализацию объекта.
         * 
         * Без аргументов – создаёт начало координат на плоскости.
         * 
         * Параметры:
         * None
         */
        public Point3D() : base(3) { }

        /*
         * Point3D – конструктор, описывающий инициализацию объекта.
         * 
         * Массив дробных чисел – создаёт точку на плоскости, значения координат берёт из массива дробных чисел.
         * 
         * Параметры:
         * double[] x
         */
        public Point3D(double[] x) : base(3, x) { }


        /*
         * (static) cross_prod – векторное произведение двух точек.
         */
        public static Point3D cross_prod(Point3D p1, Point3D p2) => new Point3D(new double[3]
            {p1.x[1] * p2.x[2] - p1.x[2] * p2.x[1],
            p1.x[2] * p2.x[0] - p1.x[0] * p2.x[2],
            p1.x[0] * p2.x[1] - p1.x[1] * p2.x[0]}
        );

        /*
         * cross_prod – векторное произведение двух точек.
         */
        public Point3D cross_prod(Point3D p)
        {
            Point3D point = cross_prod(this, p);
            x = point.x;
            dim = point.dim;
            return point;
        }

        /*
         * (static) mix_prod – смешанное произведение трёх точек.
         * 
         * (a*[b x c])
         */
        public static double mix_prod(Point3D p1, Point3D p2, Point3D p3) => mult(p1, cross_prod(p2, p3));


        /*
         * mix_prod – смешанное произведение трёх точек.
         * 
         * (a*[b x c])
         */
        public double mix_prod(Point3D p1, Point3D p2) => mix_prod(this, p1, p2);

    }
}
